### PR TITLE
fix(ui-ux): add cancel prompt function on all transaction

### PR DESCRIPTION
**Pull request by G9000**
#### What kind of PR is this?:
/kind feature

#### What this PR does / why we need it:
Adding cancel prompt functionality on the following transaction screen:

- Send
- Convert
- Add Liquidity
- Remove Liquidity
- Dex composite swap

Update the following affected e2e but does not fix the whole e2e as its kinda huge. Might want to fix on separate PR:

- Convert spec
- Remove liquidity spec
- Send spec
- Address spec

#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x] No console errors on web
- [x] Tested on Light mode and Dark mode*
- [x] Your UI implementation visually matched the rendered design*
- [x] Unit tests*
- [x] Added e2e tests*
- [x] Added translations*

### CODE REVIEW

**Comment by thedoublejay** [comment]
@G9000 I will move this to next release

**Comment by pierregee** [comment]
Bug:
1. Top up DUSD and DFI
2. Press top liquidity DUSD-DFI pool see screenshot
3. Add liquidity + continue and press cancel
4. Issue: Redirected back to Portfolio page
image

File: mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/ConfirmCompositeSwapScreen.tsx [hunk 1]
```diff
@@ -146,10 +148,29 @@ export function ConfirmCompositeSwapScreen({ route }: Props): JSX.Element {
...
+            text: translate("screens/Settings", "Go back"),
+            style: "cancel",
+          },
+          {
+            text: translate("screens/Settings", "Cancel"),
+            style: "destructive",
+            onPress: async () => {
+              navigation.navigate(
+                originScreen === DexScreenOrigin.Dex_screen
+                  ? "DexScreen"
```
Comment (pierregee) [inline mobile-app/app/screens/AppNavigator/screens/Dex/CompositeSwap/ConfirmCompositeSwapScreen.tsx:168]: Can use the enum `DexScreenOrigin.` here?

File: mobile-app/app/screens/AppNavigator/screens/Dex/DexConfirmRemoveLiquidity.tsx [hunk 2]
```diff
@@ -69,10 +77,29 @@ export function RemoveLiquidityConfirmScreen({ route }: Props): JSX.Element {
...
+            text: translate("screens/Settings", "Go back"),
+            style: "cancel",
+          },
+          {
+            text: translate("screens/Settings", "Cancel"),
+            style: "destructive",
+            onPress: async () => {
+              navigation.navigate(
+                originScreen === DexScreenOrigin.Dex_screen
+                  ? "DexScreen"
```
Comment (pierregee) [inline mobile-app/app/screens/AppNavigator/screens/Dex/DexConfirmRemoveLiquidity.tsx:97]: same here

**Review by pierregee — CHANGES_REQUESTED** [review]

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/ConvertConfirmationScreen.tsx [hunk 3]
```diff
@@ -83,12 +84,25 @@ export function ConvertConfirmationScreen({ route }: Props): JSX.Element {
...
+        buttons: [
+          {
+            text: translate("screens/Settings", "Go back"),
+            style: "cancel",
+          },
+          {
+            text: translate("screens/Settings", "Cancel"),
+            style: "destructive",
+            onPress: async () => {
+              navigation.navigate("PortfolioScreen");
```
Comment (pierregee) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/ConvertConfirmationScreen.tsx:102]: can use enum here too

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/SendConfirmationScreen.tsx [hunk 4]
```diff
@@ -102,12 +103,25 @@ export function SendConfirmationScreen({ route }: Props): JSX.Element {
...
+        buttons: [
+          {
+            text: translate("screens/Settings", "Go back"),
+            style: "cancel",
+          },
+          {
+            text: translate("screens/Settings", "Cancel"),
+            style: "destructive",
+            onPress: async () => {
+              navigation.navigate("PortfolioScreen");
```
Comment (pierregee) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/SendConfirmationScreen.tsx:121]: here too

**Review by pierregee — CHANGES_REQUESTED** [review]
image
This warning is throwing on iOS

**Review by pierregee — APPROVED** [review]

**Comment by pierregee** [comment]
1. add_liquidity.spec seems to be failing consistently after the cancel prompt
image
can just fix in another PR

**Pull request merged**